tests/optim/helpers/models.py

```diff
@@ -0,0 +1,29 @@
+from typing import Type
+
+import torch
+
+
+def _check_layer_in_model(
+    self,
+    model: torch.nn.Module,
+    layer: Type[torch.nn.Module],
+) -> None:
+    def check_for_layer_in_model(model, layer) -> bool:
+        for name, child in model._modules.items():
+            if child is not None:
+                if isinstance(child, layer):
+                    return True
+                if check_for_layer_in_model(child, layer):
+                    return True
+        return False
+
+    self.assertTrue(check_for_layer_in_model(model, layer))
+
+
+def _check_layer_not_in_model(

```

Comment:
we may not even need this `layer_not_in_model` function, just `not check_layer_in_model(model, layer)`

Reply:
@aobo-y 

In addition to using `not`, we can also use `self.assertFalse`

```
self.assertFalse(check_layer_in_model(model, layer))
self.assertTrue(check_layer_in_model(model, layer))
```